control-plane/plugin/src/resources/volume.rs

```diff
@@ -58,9 +72,9 @@ impl GetHeaderRow for openapi::models::Volume {
 }
 
 #[async_trait(?Send)]
-impl List for Volumes {
-    async fn list(output: &utils::OutputFormat) {
-        if let Some(volumes) = get_paginated_volumes().await {
+impl ListVolumes for Volumes {
```

Comment:
Instead we could have ListExt trait which has an associated context example:
```rust
trait ListExt {
type Context;
fn list(output, context: Self::Context);
}
```

Reply:
Ahh this is nice. Thanks.